scripts/Diffraction/isis_powder/osiris_routines/osiris_algs.py

```diff
@@ -220,26 +279,47 @@ def get_empty_runs(inst_settings):
     return [_assign_drange_empty("drange" + str(i + 1), all_run_numbers) for i in range(12)]
 
 
-def _correct_drange_overlap(merged_ws, drange_sets):
-    # Create scalar data to cope with where merge has combined overlapping data.
-    data_x = (merged_ws.dataX(0)[1:] + merged_ws.dataX(0)[:-1]) / 2.0
-    data_y = np.zeros(data_x.size)
-    for drange in drange_sets:
-        if drange_sets[drange].has_sample():
-            for i in range(data_x.size):
-                if d_range_alice[drange][0] <= data_x[i] <= d_range_alice[drange][1]:
-                    data_y[i] += len(drange_sets[drange].get_samples())
+def _correct_drange_overlap(merged_group_ws, drange_sets):
+    for merged_ws in merged_group_ws:
+        # Create scalar data to cope with where merge has combined overlapping data.
+        data_x = (merged_ws.dataX(0)[1:] + merged_ws.dataX(0)[:-1]) / 2.0
+        data_y = np.zeros(data_x.size)
+        for drange in drange_sets:
+            if drange_sets[drange].has_sample():
+                for i in range(data_x.size):
+                    if d_range_alice[drange][0] <= data_x[i] <= d_range_alice[drange][1]:
+                        data_y[i] += len(drange_sets[drange].get_samples())
+
+        # apply scalar data to result workspace
+        for i in range(0, merged_ws.getNumberHistograms()):
+            merged_ws.setY(i, merged_ws.dataY(i) / data_y)
+            merged_ws.setE(i, merged_ws.dataE(i) / data_y)
+    return merged_group_ws
+
+
+def merge_dspacing_runs(focussed_runs, drange_sets, run_number):
+    if len(focussed_runs) == 1:
+        return [focussed_runs[0]]
+
+    extracted_spectra = [common.extract_ws_spectra(ws) for ws in focussed_runs]
+
+    have_same_spectra_count = len({len(spectra) for spectra in extracted_spectra}) == 1
+    if not have_same_spectra_count:
+        logger.warning("Cannot merge focussed workspaces with different number of spectra")
+        return [ws for ws in focussed_runs]
 
-    # apply scalar data to result workspace
-    for i in range(0, merged_ws.getNumberHistograms()):
-        merged_ws.setY(i, merged_ws.dataY(i) / data_y)
-        merged_ws.setE(i, merged_ws.dataE(i) / data_y)
-    return merged_ws
+    # Group workspaces located at the same index
+    matched_spectra = [list(spectra) for spectra in zip(*extracted_spectra)]
+    # Merge workspaces located at the same index
+    merged_spectra = [
+        MergeRuns(InputWorkspaces=spectra, OutputWorkspace="OSIRIS" + run_number + WORKSPACE_SUFFIX.MERGED + f"_{idx}")
+        for idx, spectra in enumerate(matched_spectra)
+    ]
+    grouped_spectra = GroupWorkspaces(InputWorkspaces=merged_spectra, OutputWorkspace="OSIRIS" + run_number + WORKSPACE_SUFFIX.MERGED)
```

Comment:
I think you should be calling ConjoinSpectra here so that the merged dRanges should be placed in the same workspace giving a workspace with 8 spectra (same as the UI does)?
Also if you make this change please could you add a second system test that merges dRanges? You could write one that processes runs 199977, 199978. I think the test data files for both inputs already exist.

Reply:
Ok, I will do these changes

Reply:
The ConjoinSpectra change looks fine. Please can you add the extra system test that uses `merge_drange=True`?

Reply:
Yeah, I already did but there is a mismatch problem